hilti/runtime/include/types/regexp.h

```diff
@@ -32,6 +32,9 @@ struct Flags {
 
     friend bool operator==(const Flags& a, const Flags& b) { return a.no_sub == b.no_sub && a.use_std == b.use_std; }
     friend bool operator!=(const Flags& a, const Flags& b) { return ! (a == b); }
+
+    /** Returns a string representation of the flags that can be used as part of a cache key. */
+    std::string cacheKey() const { return fmt("%c%c", no_sub ? '1' : '0', use_std ? '1' : '0'); }
```

Comment:
Not sure I see how this would get much better that way because we'd still need to combine it with a string consisting of all the patterns. But I'll make it a bit more efficient at least.

Reply:
But we're building a string as the map index, so we'd either need to convert the hash into a string representation then, or use the hash itself as the map key, but that doesn't seem right.

Reply:
This I kept even in the new version, because I couldn't find a good approach to hashing the `shared_ptr` that we store in the global map.